rs/tests/driver/src/nns.rs

```diff
@@ -490,17 +491,24 @@ pub async fn submit_update_elected_replica_versions_proposal(
         neuron_id,
         NnsFunction::ReviseElectedGuestosVersions,
         ReviseElectedGuestosVersionsPayload {
-            replica_version_to_elect: Some(String::from(version.clone())),
-            release_package_sha256_hex: Some(sha256.clone()),
+            replica_version_to_elect: version.clone().map(String::from),
+            release_package_sha256_hex: sha256.clone(),
             release_package_urls: upgrade_urls,
-            replica_versions_to_unelect: versions_to_unelect,
+            replica_versions_to_unelect: versions_to_unelect.clone(),
             guest_launch_measurement_sha256_hex: None,
         },
-        format!(
-            "Elect replica version: {} with hash: {}",
-            String::from(version),
-            sha256
-        ),
+        match (version, sha256, versions_to_unelect.is_empty()) {
+            (Some(v), Some(sha), _) => format!(
+                "Elect replica version: {} with hash: {}",
+                String::from(v),
+                sha
+            ),
+            (None, None, false) => format!(
+                "Retiring versions: {}",
```

Comment:
Are versions specified in `versions_to_unelect` also retired in case `version` and `sha256` are set to some values? If so, only describing `"Retiring versions: {}"` in case `versions` and `sha256` are none seems incomplete.

Reply:
They are, but I am not sure if these proposals are going to be read at all? I can adapt if needed of course